zebra-chain/src/addresses.rs

```diff
@@ -0,0 +1,222 @@
+//! Address types.
+
+use std::{fmt, io};
+
+use bs58;
+use ripemd160::{Digest, Ripemd160};
+use secp256k1::PublicKey;
+use sha2::Sha256;
+
+#[cfg(test)]
+use proptest_derive::Arbitrary;
+
+use crate::serialization::{SerializationError, ZcashDeserialize, ZcashSerialize};
+
+use crate::types::Script;
+
+/// A hash of a redeem script, as used in transparent
+/// pay-to-script-hash and pay-to-publickey-hash addresses.
+///
+/// The resulting hash in both of these cases is always exactly 20
+/// bytes. These are big-endian.
+/// https://en.bitcoin.it/Base58Check_encoding#Encoding_a_Bitcoin_address
+#[derive(Copy, Clone, Eq, PartialEq)]
+#[cfg_attr(test, derive(Arbitrary))]
+pub struct AddressPayloadHash(pub [u8; 20]);
+
+impl<'a> From<&'a [u8]> for AddressPayloadHash {
+    fn from(bytes: &'a [u8]) -> Self {
+        let sha_hash = Sha256::digest(bytes);
+        let ripe_hash = Ripemd160::digest(&sha_hash);
+        let mut payload = [0u8; 20];
+        payload[..].copy_from_slice(&ripe_hash[..]);
+        Self(payload)
+    }
+}
+
+impl fmt::Debug for AddressPayloadHash {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_tuple("AddressPayloadHash")
+            .field(&hex::encode(&self.0))
+            .finish()
+    }
+}
+
+/// An enum describing the possible network choices.
+// XXX Stolen from zebra-network for now.
```

Comment:
Since this enum is needed for chain structures, I think it would probably be better to move the definition from zebra-network into, say, `zebra-chain/src/network.rs`, reëxporting it from the crate root, and then updating zebra-network to use `zebra_chain::Network` everywhere.

Reply:
Should I bring the `magics` in `zebra-network/src/constants.rs` along too?